src/main/java/com/celements/web/service/CelementsWebService.java

```diff
@@ -74,60 +85,92 @@ public int createUser(boolean validate) throws XWikiException {
   @Override
   public synchronized int createUser(Map<String, String> userData, String possibleLogins,
       boolean validate) throws XWikiException {
+    try {
+      if (createNewUser(userData, possibleLogins, validate) != null) {
+        return 1;
+      }
+    } catch (UserCreateException uce) {
+      if (uce.getCause() instanceof XWikiException) {
+        throw (XWikiException) uce.getCause();
+      } else {
+        throw new XWikiException(XWikiException.MODULE_XWIKI_STORE,
+            XWikiException.ERROR_XWIKI_UNKNOWN, "failed to create user", uce);
+      }
+    }
+    return -1;
+  }
+
+  @Override
+  public synchronized @NotNull XWikiUser createNewUser(@NotNull Map<String, String> userData,
+      @NotNull String possibleLogins, @NotNull boolean validate) throws UserCreateException {
```

Comment:
I just added @NotNull to validate to be consistent, but I think it makes sense to add those annotations for nullable types only.